src/multi-select/multi-select.component.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { PropTypes } from 'react';
+import MultiSelectItem from '../multi-select-item/multi-select-item.component';
+import './multi-select.component.scss';
+
+export default class MultiSelect extends React.PureComponent {
+
+  static propTypes = {
+    items: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+      ]).isRequired,
+      text: PropTypes.string.isRequired,
+    })).isRequired,
+    checkedItems: PropTypes.arrayOf(
+      PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+      ]),
+    ).isRequired,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    onChange: () => {},
+  };
+
+  shouldComponentUpdate(nextProps) {
+    if (String(this.props.checkedItems) !== String(nextProps.checkedItems)) {
```

Comment:
Converting a long list into a String and then performing strict compare is not as efficient as shallow comparison using object reference which the PureComponent extend does by default.